pkg/collector/corechecks/servicediscovery/impl_linux.go

```diff
@@ -139,23 +107,7 @@ func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
 		if _, ok := li.aliveServices[pid]; !ok {
 			log.Debugf("[pid: %d] found new process with open ports", pid)
 
-			p, ok := procs[pid]
-			if !ok {
-				log.Debugf("[pid: %d] process with open ports was not found in alive procs", pid)
-				continue
-			}
-
-			svc, err := li.getServiceInfo(p, service)
-			if err != nil {
-				telemetryFromError(errWithCode{
-					err:  err,
-					code: errorCodeProcfs,
-					svc:  nil,
-				})
-				log.Errorf("[pid: %d] failed to get process info: %v", pid, err)
-				li.ignoreProcs[pid] = true
-				continue
-			}
+			svc := li.getServiceInfo(pid, service)
```

Comment:
`svc` is a pointer, to be future-proof - please check it is not nil before accessing to its attributes 

Also, why do we return a pointer instead of a value?

Reply:
Done, removed pointer return.